bench-vortex/benches/compress_benchmark.rs

```diff
@@ -1,50 +1,236 @@
+use std::fs;
+use std::path::Path;
+use std::time::Duration;
+
+use arrow_array::RecordBatch;
 use bench_vortex::data_downloads::BenchmarkDataset;
 use bench_vortex::public_bi_data::BenchmarkDatasets;
 use bench_vortex::public_bi_data::PBIDataset::*;
 use bench_vortex::taxi_data::taxi_data_parquet;
 use bench_vortex::tpch::dbgen::{DBGen, DBGenOptions};
-use bench_vortex::{compress_taxi_data, tpch};
-use criterion::{black_box, criterion_group, criterion_main, Criterion};
-use vortex::{IntoArray, IntoCanonical};
+use bench_vortex::{fetch_taxi_data, tpch};
+use criterion::{
+    black_box, criterion_group, criterion_main, BenchmarkGroup, Criterion, Throughput,
+};
+use parquet::arrow::ArrowWriter;
+use parquet::basic::{Compression, ZstdLevel};
+use parquet::file::properties::WriterProperties;
+use tokio::fs::File;
+use tokio::io::{AsyncSeekExt, AsyncWriteExt};
+use vortex::array::{ChunkedArray, StructArray};
+use vortex::{Array, ArrayDType, IntoArray, IntoCanonical};
+use vortex_dtype::field::Field;
 use vortex_sampling_compressor::compressors::fsst::FSSTCompressor;
 use vortex_sampling_compressor::SamplingCompressor;
+use vortex_serde::layouts::LayoutWriter;
+
+#[derive(serde::Serialize)]
+struct GenericBenchmarkResults<'a> {
+    name: &'a str,
+    value: f64,
+    unit: &'a str,
+    range: f64,
+}
+
+fn ensure_dir_exists(dir: &str) -> std::io::Result<()> {
+    let path = Path::new(dir);
+    if !path.exists() {
+        fs::create_dir_all(path)?;
+    }
+    Ok(())
+}
+
+fn parquet_written_size(array: &Array, filepath: &str, compression: Compression) -> usize {
+    let mut file = std::fs::File::create(Path::new(filepath)).unwrap();
+    let chunked = ChunkedArray::try_from(array).unwrap();
+    let chunks_vec = chunked.chunks().collect::<Vec<_>>();
+
+    if chunks_vec.is_empty() {
+        panic!("empty chunks");
+    }
+
+    let schema = RecordBatch::try_from(chunks_vec[0].clone())
+        .unwrap()
+        .schema();
+
+    let writer_properties = WriterProperties::builder()
+        .set_compression(compression)
+        .build();
+    let mut writer = ArrowWriter::try_new(&mut file, schema, Some(writer_properties)).unwrap();
+    for chunk in chunks_vec {
+        let record_batch = RecordBatch::try_from(chunk).unwrap();
+        writer.write(&record_batch).unwrap();
+    }
+    writer.flush().unwrap();
+    let n_bytes = writer.bytes_written();
+    writer.close().unwrap();
+    n_bytes
+}
+
+fn vortex_written_size(array: &Array, filepath: &str) -> u64 {
```

Comment:
same thing here: you can use a Vec<u8> instead of file if all you wanna do is measure the compressed size

Reply:
though i could understand if you wanna dump them to disk to poke at manually, persisting would make that easier

Reply:
Yeah, I did poke at them while I was iterating but at least for testing and benchmarking it seems best to be fast and not use disk.